src/main/java/uk/nhs/digital/nhsconnect/nhais/service/RegistrationConsumerService.java

```diff
@@ -2,56 +2,65 @@
 public class RegistrationConsumerService {
 
     private final InboundGpSystemService inboundGpSystemService;
     private final InboundStateRepository inboundStateRepository;
+    private final OutboundStateRepository outboundStateRepository;
+    private final OutboundMeshService outboundMeshService;
+    private final RecepProducerService recepProducerService;
+    private final EdifactParser edifactParser;
 
     public void handleRegistration(MeshMessage meshMessage) {
         LOGGER.debug("Received Registration message: {}", meshMessage);
-        Interchange interchange;
-        try {
-            interchange = new EdifactParser().parse(meshMessage.getContent());
-        } catch (ToEdifactParsingException ex) {
-            LOGGER.error("Errors during parsing MESH message into EDIFACT", ex);
-            throw ex;
-        }
+        Interchange interchange = edifactParser.parse(meshMessage.getContent());
         LOGGER.debug("Parsed registration message into interchange: {}", interchange);
 
         var inboundState = InboundState.fromInterchange(interchange);
-        if (!saveState(inboundState)) {
+        if (!saveInboundState(inboundState)) {
             return;
         }
 
-        sendSequenceNumbersToInboundSequenceManager(inboundState); //TODO implementation of that
+        var recep = recepProducerService.produceRecep(interchange);
+        var recepOutboundState = OutboundState.fromRecep(recep);
+        outboundStateRepository.save(recepOutboundState);
+        LOGGER.debug("Saved recep in outbound state: {}", recepOutboundState);
 
-        // recep producer service
-        Parameters outputParameters = new EdifactToFhirService().convertToFhir(interchange);
+        var outputParameters = new EdifactToFhirService().convertToFhir(interchange);
         LOGGER.debug("Converted registration message into FHIR: {}", outputParameters);
         inboundGpSystemService.publishToSupplierQueue(outputParameters, inboundState.getOperationId());
         LOGGER.debug("Published inbound registration message to gp supplier queue");
+
+        var recepMeshMessage = buildRecepMeshMessage(recep);
+        LOGGER.debug("Wrapped recep in mesh message: {}", recepMeshMessage);
+        outboundMeshService.publishToOutboundQueue(recepMeshMessage);
+        LOGGER.debug("Published recep to outbound queue");
     }
 
-    private void sendSequenceNumbersToInboundSequenceManager(InboundState inboundState) {
-        //sequence numbers for Inbound Sequence Number Manager can be extracted from InboundState
-        LOGGER.debug("Interchange sequence number: " + inboundState.getReceiveInterchangeSequence());
-        LOGGER.debug("Message sequence number: " + inboundState.getReceiveMessageSequence());
+    private MeshMessage buildRecepMeshMessage(Recep recep) {
+        return new MeshMessage()
+            // TODO: determine ODS code: probably via ENV? or should it be taken from incoming mesh message?
+            .setOdsCode("ods123")
```

Comment:
can we leave ods123 for now?

Reply:
it's the same hardcoded value as used in other place
we don't have ods code support now

Reply:
It won't be Ods code anymore. We'll use the recipient cipher. Can change now to we'll do it when we implement send via mesh